GliaWidgetsTests/SecureConversations/Welcome/SecureConversations.WelcomeViewModelSpec.swift

```diff
@@ -190,7 +215,489 @@ extension SecureConversationsWelcomeViewModelTests {
+    }
+
+    func testMediaPickerDoesNotCallBrowseUploadCallback() {
+        viewModel.fileUploadListModel.environment.uploader = FileUploader(maximumUploads: 100, environment: .mock)
+
+        viewModel.delegate = { event in
+            switch event {
+            case let .mediaPickerRequested(_, callback):
+                callback(.browse)
+            case .pickFile(let callback):
+                callback(.cancelled)
+            default: break
+            }
+        }
+
+        viewModel.presentMediaPicker(from: UIView(), alertConfiguration: viewModel.environment.alertConfiguration)
+
+        XCTAssertTrue(viewModel.fileUploadListModel.activeUploads.isEmpty)
+    }
+}
+
+extension SecureConversationsWelcomeViewModelTests {
+    func testAvailabilityAvailable() {
+        let uuid = UUID().uuidString
+        var availability = SecureConversations.Availability.mock
+        availability.environment.listQueues = { completion in
+            let queue = CoreSdkClient.Queue(
+                id: uuid,
+                name: "",
+                status: .open,
+                isDefault: true,
+                media: [.messaging]
+            )
+            completion([queue], nil)
+        }
+
+        availability.environment.isAuthenticated = { true }
+        availability.environment.queueIds = [uuid]
+
+        viewModel = .init(environment: .mock, availability: availability)
+
+        XCTAssertEqual(viewModel.availabilityStatus, .available)
+    }
+
+    func testAvailabilityUnavailableEmptyQueues() {
+        var alertConfiguration: MessageAlertConfiguration?
+
+        let uuid = UUID().uuidString
+        var availability = SecureConversations.Availability.mock
+        availability.environment.listQueues = { completion in
+            let queue = CoreSdkClient.Queue(
+                id: uuid,
+                name: "",
+                status: .open,
+                isDefault: true,
+                media: [.text]
+            )
+            completion([queue], nil)
+        }
+
+        availability.environment.isAuthenticated = { true }
+        availability.environment.queueIds = [uuid]
+
+        let delegate: (WelcomeViewModel.DelegateEvent) -> Void = { event in
+            switch event {
+            case .showAlertAsView(let configuration, _, _):
+                alertConfiguration = configuration
+            default: break
+            }
+        }
+
+        viewModel = .init(
+            environment: .mock,
+            availability: availability,
+            delegate: delegate
+        )
+
+        XCTAssertEqual(viewModel.availabilityStatus, .unavailable(.emptyQueue))
+        XCTAssertEqual(
+            alertConfiguration?.message,
+            viewModel.environment.alertConfiguration.unavailableMessageCenter.message
+        )
+    }
+
+    func testAvailabilityUnavailableUnauthenticated() {
+        var alertConfiguration: MessageAlertConfiguration?
+
+        let uuid = UUID().uuidString
```

Comment:
N. B. I'd suggest to use UUID.mock to encourage same results in tests. For example if we are to use text-based snapshot test it will be noticeable right away.

Reply:
Fixed b3af2943f357e79a5759a7110a5d31c334eafaf9